starfish/test/full_pipelines/api/test_dartfish.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import pandas as pd
+
+from starfish import Codebook, Experiment
+from starfish.image._filter.scale_by_percentile import ScaleByPercentile
+from starfish.image._filter.zero_by_channel_magnitude import ZeroByChannelMagnitude
+from starfish.spots._detector.pixel_spot_detector import PixelSpotDetector
+from starfish.types import Features
+
+
+def test_dartfish_pipeline_cropped_data():
+
+    # load the experiment
+    experiment_json = (
+        'https://dmf0bdeheu4zf.cloudfront.net/'
+        'TEST/20180821/DARTFISH_TEST/fov_001/experiment.json'
+    )
+    experiment = Experiment.from_json(experiment_json)
+
+    primary_image = experiment.image
+    primary_image._data = primary_image._data.astype(float)
+
+    expected_primary_image = np.array(
+        [[1., 0., 0., 0., 0., 0., 3., 0., 0., 2.],
+         [0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+         [0., 0., 0., 0., 1., 0., 0., 0., 2., 0.],
+         [0., 0., 1., 0., 0., 1., 0., 1., 0., 0.],
+         [0., 0., 0., 0., 0., 0., 2., 0., 1., 0.],
+         [0., 0., 1., 0., 2., 0., 0., 0., 0., 1.],
+         [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],
+         [0., 1., 0., 0., 1., 0., 0., 0., 0., 0.],
+         [1., 1., 0., 0., 0., 0., 0., 0., 0., 0.],
+         [0., 0., 0., 0., 0., 1., 1., 0., 0., 0.]],
+        dtype=np.float
+    )
+
+    assert np.array_equal(
+        primary_image.numpy_array[0, 0, 0, 50:60, 60:70],
+        expected_primary_image
+    )
+
+    sc_filt = ScaleByPercentile(p=100)
+    normalized_image = sc_filt.run(primary_image, in_place=False)
+
+    expected_normalized_image = np.array(
+        [[0.01960784, 0., 0., 0., 0.,
+          0., 0.05882353, 0., 0., 0.03921569],
+         [0., 0., 0., 0., 0.01960784,
+          0., 0., 0., 0., 0.],
+         [0., 0., 0., 0., 0.01960784,
+          0., 0., 0., 0.03921569, 0.],
+         [0., 0., 0.01960784, 0., 0.,
+          0.01960784, 0., 0.01960784, 0., 0.],
+         [0., 0., 0., 0., 0.,
+          0., 0.03921569, 0., 0.01960784, 0.],
+         [0., 0., 0.01960784, 0., 0.03921569,
+          0., 0., 0., 0., 0.01960784],
+         [0., 0., 0., 0., 0.,
+          0.01960784, 0., 0., 0., 0.],
+         [0., 0.01960784, 0., 0., 0.01960784,
+          0., 0., 0., 0., 0.],
+         [0.01960784, 0.01960784, 0., 0., 0.,
+          0., 0., 0., 0., 0.],
+         [0., 0., 0., 0., 0.,
+          0.01960784, 0.01960784, 0., 0., 0.]])
```

Comment:
does this need a dtype?

Reply:
float is the default dtype, but I will make all such cases explicit.